PO- 1450 Added Capitalisation Directive + Unit tests (#45)
hmcts/opal-frontend-common-ui-lib: projects/opal-frontend-common/directives/capitalisation/readme.md

Reviewer: I wonder if it is worth adding a section to the readme to talk about the selector 

Selector
`[opalLibCapitaliseAllCharacters]`

⸻
Capitalisation Directive

This Angular directive automatically capitalises all characters in an input field in real-time as the user types. It is used to ensure consistent formatting in fields like reference numbers or codes.

Table of Contents
• Installation
• Usage
• Inputs
• Outputs
• Methods
• Testing

> • Contributing

Reviewer: Watch the table of contents if you are changing stuff

# Capitalisation Directive

This Angular directive automatically capitalises all characters in an input field in real-time as the user types. It is used to ensure consistent formatting in fields like reference numbers or codes.

## Table of Contents

- [Installation](#installation)
- [Selector](#selector)
- [Usage](#usage)
- [Inputs](#inputs)
- [Outputs](#outputs)
- [Methods](#methods)
- [Testing](#testing)
- [Contributing](#contributing)

## Installation

> ```typescript
> Import { CapitalisationDirective } from '@hmcts/opal-frontend-common/directives/capitalisation';
> ```

Reviewer: ```suggestion
```typescript
import { CapitalisationDirective } from '@hmcts/opal-frontend-common/directives/capitalisation';
```
```

Ensure the directive is declared or imported in your component/module:

```typescript
@Component({
  imports: [CapitalisationDirective],
})
export class SharedModule {}
```

## Selector

> Use this attribute selector on an <input> element or a custom component that emits the native input event. The directive listens for input events and capitalises the input value in real time.

Reviewer: ```suggestion
Use this attribute selector on an `<input>` element or a custom component that emits the native input event. The directive listens for input events and capitalises the input value in real time.
```

> ```md
> [opalLibCapitaliseAllCharacters]
> ```

Reviewer: ```suggestion
```
[opalLibCapitaliseAllCharacters]
```
```

## Usage

> You can apply the directive to any component or element that contains an <input> element inside it. The directive listens to input events and converts the value to uppercase.

Reviewer: ```suggestion
You can apply the directive to any component or element that is an `<input>` element inside it. The directive listens to input events and converts the value to uppercase.
```

Reviewer: Maybe change it to this instead
```suggestion
You can apply the directive to any `<input>` element or a custom component that emits an `input` event. The directive listens for input events and converts the value to uppercase.
```

⸻

Capitalisation Directive

This Angular directive automatically capitalises all characters in an input field in real-time as the user types. It is used to ensure consistent formatting in fields like reference numbers or codes.

Table of Contents
• Installation
• Usage
• Inputs
• Outputs
• Methods
• Testing
• Contributing

⸻

Installation

import { CapitalisationDirective } from '@hmcts/opal-frontend-common/directives/capitalisation';

Ensure the directive is declared in your component:

@Component({
imports: [CapitalisationDirective]
})
export class SharedModule {}

⸻

Usage

You can apply the directive to any component or element that contains an <input> element inside it. The directive listens to the input event and transforms the value to uppercase.

Example Usage with Native Input:

<div capitaliseAllCharacters>
  <input type="text" />

> </div>

Reviewer: The directive binds to the input event on the element the directive is applied to. It does not query or traverse child elements to find an <input>. Therefore this can be removed

Example Usage with Custom Component:

<opal-lib-govuk-text-input
capitaliseAllCharacters
inputId="exampleInput"
inputName="exampleInput"
labelText="Reference Number"
[control]="form.controls['exampleInput']"

> />

Reviewer: This method is correct. However, it uses the wrong directive name. Can you update it?
